dev/io.openliberty.mcp.internal_fat/fat/src/io/openliberty/mcp/internal/fat/tool/AsyncToolsTest.java

```diff
@@ -301,4 +306,81 @@ public void testCompletionStageThatNeverCompletes() throws Exception {
                         """;
         client.callMCP(request);
     }
+
+    @SuppressWarnings({ "unchecked", "serial" })
+    @Test
+    public void testAsyncObjectToolHasExpectedOutputSchema() throws Exception {
+        String response = client.callMCP("""
+                            {
+                              "jsonrpc": "2.0",
+                              "id": 1,
+                              "method": "tools/list"
+                            }
+                        """);
+
+        Jsonb jsonb = JsonbBuilder.create();
+        Map<String, Object> parsed = jsonb.fromJson(response, new TypeLiteral<Map<String, Object>>() {}.getType());
+        Map<String, Object> result = (Map<String, Object>) parsed.get("result");
+        List<Map<String, Object>> tools = (List<Map<String, Object>>) result.get("tools");
+
+        Map<String, Object> asyncObjectTool = null;
```

Comment:
We would need to add Jsonb impl to build.gradle to do this.

But!

We can also use the JSONAssert library that the tests already use to do the same thing.

See https://jsonassert.skyscreamer.org/apidocs/org/skyscreamer/jsonassert/JSONParser.html to parse the response, find the bin you're interested in and then use JSONAssert as before to assert it matches what you exepect.